mkat_fpga_tests/test_cbf.py

```diff
@@ -692,52 +737,52 @@ def test_sensor_values(self):
+    @aqf_vr('TP.C.1.16')
     def test_roach_qdr_sensors(self):
-        """ """
-        an_e = threading.Event()
-        def event_(an_e, *args):
-            print 'Event occured'
-            try:
-                an_e.set()
-            except Exception, exc:
-                print exc
-        an_event = partial(event_, an_e)
-
         array_sensors = correlator_fixture.katcp_rct.sensor
+        # Select a host
         xhost = self.correlator.xhosts[0]
-        xhost.blindwrite('qdr1_memory', 'write_junk_to_memory')
-        Aqf.is_true(
-            array_sensors.roach020a0a_xeng_qdr.get_value() == xhost.qdr_okay(),
-                'Check that the memory is corrupted.')
-
-        Aqf.is_true(array_sensors.roach020a0a_xeng_qdr.get_value(),
-            'Check that the memory recovered successfully.')
-        array_sensors.roach020a0a_xeng_qdr.set_strategy('auto')
-        array_sensors.roach020a0a_xeng_qdr.register_listener(an_event)
-
+        Aqf.step("Selected host: {}".format(xhost.host))
+        # Check if qdr is okay
+        Aqf.is_true(xhost.qdr_okay(), 'Check that the QDR has not failed.')
+        Aqf.step("Writing junk to {} memory.".format(xhost.host))
+        # Write junk to memory
+
+        for i in range(5):
+            xhost.blindwrite('qdr1_memory', 'write_junk_to_memory')
+            Aqf.wait(.1,'Wait before checking is memory is corrupted.')
+        # Verify that qdr corrupted or unreadable
+        Aqf.is_false(xhost.qdr_okay(),
+            'Check that the memory on {} is unreadable/corrupted.'.format(xhost.host))
+        current_errors = xhost.registers.vacc_errors1.read()['data']['parity']
+        Aqf.is_not_equals(current_errors, 0, "Error counters are incrementing.")
+        if current_errors == xhost.registers.vacc_errors1.read()['data']['parity']:
+            Aqf.passed('Confirm that the counters have stopped incrementing: {} increments.'
+                .format(current_errors))
+        else:
+            Aqf.failed('Error counters still incrementing.')
+        # Check that the memory recovered successfully
         Aqf.is_true(array_sensors.roach020a0a_xeng_qdr.get_value(),
             'Check that the memory recovered successfully.')
+        # Clear and confirm error counters
+        xhost.clear_status()
+        final_errors = xhost.registers.vacc_errors1.read()['data']['parity']
+        Aqf.is_false(final_errors,
+            'Confirm that the counters have been reset, count {} to {}'
+                .format(current_errors, final_errors))
+        Aqf.is_true(xhost.qdr_okay(), 'Check that the QDR is okay.')
 
-        xhost.vacc_get_error_detail()[1]['parity']
-
-        self.addCleanup(array_sensors.roach020a0a_xeng_qdr.unregister_listener(an_event))
-        import IPython;IPython.embed()
-
-
-    @aqf_vr('TP.C.dummy_vr_6')
+    @aqf_vr('TP.C.1.16')
     def test_roach_pfb_sensors(self):
         array_sensors = correlator_fixture.katcp_rct.sensor
+        pass
```

Comment:
Perhaps make the test fail with a 'Not implmented yet' or something, otherwise the test report will lie.